### PR TITLE
feat: add an input parser

### PR DESCRIPTION
**Pull request by ppiques**
added a few things dw you can merge trust me (:

### CODE REVIEW

**Review by theovgl — APPROVED** [review]
LGTM, les tests sont blindax

File: src/IRCMessageStream.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	}
+	j = 0;
+	if (trailingArg == true && _argCount < 15)
+	{
+		it++;
+		while (it != this->_input.end() && it[j] != '\r')
+			j++;
+		this->_arguments[_argCount] = this->_input.substr(it - this->_input.begin(), j);
+		_argCount++;
+	}
```
Comment (vfurmane) [inline src/IRCMessageStream.cpp]: This should be done inside the loop above.

**Review by vfurmane — CHANGES_REQUESTED** [review]
Are you fucking brain dead @ppiques ??

File: include/IRCMessageStream.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+	private:
+
+	void	parseArguments(std::string::const_iterator &it);
+
+	std::string		_input;
+	std::string		*_prefix;
+	std::string		_command;
+	std::string		_arguments[15];
+	int				_argCount;
```
[the remark this answers is not in the document]
Comment (ppiques) [reply include/IRCMessageStream.hpp]: or it couldn't

**Review by vfurmane — APPROVED** [review]
ENFINNN

**Pull request merged**